crates/foundry-ui/src/components/tagged.rs

```diff
@@ -0,0 +1,40 @@
+use console::Style;
+use serde::Serialize;
+
+use crate::Message;
+
+/// Generic textual message with `tag` prefix.
+///
+/// The type prefix can be stylized in text mode.
+/// e.g. "[WARNING]: An example warning message"
+#[derive(Serialize)]
+pub struct TaggedMessage<'a> {
+    tag: &'a str,
+    text: &'a str,
+    #[serde(skip)]
+    type_style: Option<&'a str>,
+}
+
+impl<'a> TaggedMessage<'a> {
+    #[must_use]
+    pub fn styled(tag: &'a str, text: &'a str, type_style: &'a str) -> Self {
+        Self {
+            tag,
+            text,
+            type_style: Some(type_style),
+        }
+    }
+}
```

Comment:
In which cases `type_style` will be able to be none if we don't allow creating such instances?

Reply:
Added `raw()` method, which doesn't have `tyg_style` param, then it's `None`.